From 95f14b58f92a672669b41f1e4ce6c02f82155eca Mon Sep 17 00:00:00 2001
From: Alex Wilson <alex.wilson@joyent.com>
Date: Fri, 22 Jun 2018 15:04:29 -0700
Subject: [PATCH] MANTA-3785 waferlock allowing destroyed VMs MANTA-3786
 waferlock doesn't deal with old VMs with no "ips" property on NICs

---
 lib/sapi.js | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/lib/sapi.js b/lib/sapi.js
index c9e4ef2..25915be 100644
--- a/lib/sapi.js
+++ b/lib/sapi.js
@@ -249,10 +249,15 @@ SapiPoller.prototype.state_runq_do = function (S) {
 		mod_assert.arrayOfObject(objs);
 
 		objs.forEach(function (vm) {
-			var ips = [];
+			if (vm.state === 'destroyed' || vm.destroyed)
+				return;
 
+			var ips = [];
 			vm.nics.forEach(function (nic) {
-				nic.ips.forEach(function (ip) {
+				var ipa = nic.ips;
+				if (ipa === undefined)
+					ipa = [nic.ip];
+				ipa.forEach(function (ip) {
 					var parts = ip.split('/');
 					ips.push(parts[0]);
 				});
-- 
2.21.0

